### DISCUSSION TITLE
How to get/check references in an array?

### DISCUSSION THREAD

**Discussion by ghost** [gdsfactory/gdsfactory#3337] [Q&A] [answered]
I want to create an specific shape for an array of a repeating component, but I do not want to chop a component, but remove it if it does not fall within the specific shape. Is that possible, or do I need to explicitly place each reference?

How do I parse an array to check each reference? I do not understand how to iterate through references in a component.

```python
radius = 20
boundary_radius = 700
columns = 10
rows = 10
spacex = 80
spacey = 80

#ref circle
circle = gf.components.circle(radius=radius, angle_resolution=2.5, layer=(1, 0)) 
#create an array
circle_array = gf.components.array(circle, columns = columns, rows = rows, spacing = [spacex,spacey]) 
#define boundary
boundary = gf.components.circle(radius=boundary_radius, angle_resolution=2.5, layer=(1, 0))
#exclude circles not inside of boundary
circles = gf.boolean(A=circle_array, B=boundary, operation="and")

circles.plot()
```

**Comment by sebastian-goeldi** [comment, marked as answer]
You can't chop out individual indexes of an array, a gds AREF doesn't work that way, it's all or nothing (at least to my knowledge).

You can just iterate through the instances like you would in any list `for inst in c.insts`

There are some tricks you could play, but it goes a bit beyond the scope of what you are asking.

Anyway, I think the most suitable approach for you is to look at the filling algorithms, as I assume they do exactly what you want to do https://gdsfactory.github.io/gdsfactory/notebooks/04_components_geometry.html#dummy-fill-tiling

**Reply by ghost** [reply]
Indeed! Thanks for a quick response pointing out the exact documented answer.
